delivery_ups_oca/models/delivery_carrier.py

```diff
@@ -1,22 +1,285 @@
 # Copyright 2020 Hunki Enterprises BV
+# Copyright 2021 Tecnativa - Víctor Martínez
 # License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
-from odoo import fields, models
+from odoo import _, fields, models
+from odoo.exceptions import UserError
+
+import requests
 
 
 class DeliveryCarrier(models.Model):
     _inherit = "delivery.carrier"
 
-    delivery_type = fields.Selection(selection_add=[("ups", "UPS")],)
+    delivery_type = fields.Selection(selection_add=[("ups", "UPS")])
+    ups_file_format = fields.Selection(
+        selection=[("GIF", "PDF"), ("ZPL", "ZPL"), ("EPL", "EPL"), ("SPL", "SPL")],
+        default="GIF",
+        string="File format",
+    )
+    ups_ws_username = fields.Char(string="Username WS")
+    ups_ws_password = fields.Char(string="Password WS")
+    ups_access_license = fields.Char(string="Access license number")
+    ups_shipper_number = fields.Char(string="Shipper number")
+    ups_service_code = fields.Selection(
+        selection=[
+            ("01", "Next Day Air"),
+            ("02", "2nd Day Air"),
+            ("03", "Ground"),
+            ("07", "Express"),
+            ("08", "Expedited"),
+            ("11", "UPS Standard"),
+            ("12", "3 Day Select"),
+            ("13", "Next Day Air Saver"),
+            ("14", "UPS Next Day Air® Early"),
+            ("17", "UPS Worldwide Economy DDU"),
+            ("54", "Express Plus"),
+            ("59", "2nd Day Air A.M."),
+            ("65", "UPS Saver"),
+            ("M2", "First Class Mail"),
+            ("M3", "Priority Mail"),
+            ("M4", "Expedited Mail Innovations"),
+            ("M5", "Priority Mail Innovations"),
+            ("M6", "Economy Mail Innovations"),
+            ("M7", "Mail Innovations (MI) Returns"),
+            ("70", "UPS Access PointTM Economy"),
+            ("71", "UPS Worldwide Express Freight Midday"),
+            ("72", "UPS Worldwide Economy"),
+            ("74", "UPS Express®12:00"),
+            ("82", "UPS Today Standard"),
+            ("83", "UPS Today Dedicated Courier"),
+            ("84", "UPS Today Intercity"),
+            ("85", "UPS Today Express"),
+            ("86", "UPS Today Express Saver"),
+            ("96", "UPS Worldwide Express Freight"),
+        ],
+        default="11",
+        string="Service code",
+    )
+    ups_default_packaging_id = fields.Many2one(
+        comodel_name="product.packaging", string="Default Packaging Type"
+    )
+    ups_package_dimension_code = fields.Char(string="Package Dimension code")
+    ups_package_weight_code = fields.Char(string="Package Weight code")
+
+    def _ups_set_url(self, ups_type):
```

Comment:
This is not setting anything:

```suggestion
def _ups_get_url(self, ups_type):
```